packages/web-components/fast-foundation/src/flipper/flipper.template.ts

```diff
@@ -18,10 +18,11 @@ export function flipperTemplate<T extends FASTFlipper>(
         let existing = templateCache[direction];
 
         if (!existing) {
+            const iconKey = direction + "Icon";
             templateCache[direction] = existing = html`
```

Comment:
@chrisdholt or @EisenbergEffect can you help me understand how turning this into a function results in more efficiency? It's still a conditional render, right? And generally the same Flipper doesn't change from `next` to `previous`. And from a code size perspective, the template was rather small, so the overhead of the function seems like it would negate it even once minimized. I'm considering when to use the pattern, but don't know why it's better for smaller occurrences.

Reply:
Further thoughts, maybe it's only a slight improvement in template byte size, but that's enough to warrant it. Also, wondering about the cache since it's local to the function - perhaps the template processor is caching the function somehow, but I'm not sure how that would work, say across instances of Flippers.

Reply:
If I recall, the original template had TWO `when` directives. So, this change enables there to be a single conditional render instead of two conditional renders. The implementation uses the function in order to cache the templates so that they aren't re-created (and re-compiled) every time they are needed. `when` caches the template internally. Since we aren't doing that, we need to cache it ourself. We can't cache it as a module level variable though, because it's dependent on the specific options.